src/wallet/index.ts

```diff
@@ -0,0 +1,37 @@
+import { BitcoinWallet } from "./bitcoin";
+import { EthereumWallet } from "./ethereum";
+import { LightningWallet } from "./lightning";
+
+export interface AllWallets {
+  bitcoin?: BitcoinWallet;
+  ethereum?: EthereumWallet;
+  lightning?: LightningWallet;
+}
+
+export class Wallets {
+  constructor(private readonly wallets: AllWallets) {}
+
+  get bitcoin(): BitcoinWallet {
+    return this.getWalletForLedger("bitcoin") as BitcoinWallet;
+  }
+
+  get ethereum(): EthereumWallet {
+    return this.getWalletForLedger("ethereum") as EthereumWallet;
+  }
+
+  get lightning(): LightningWallet {
+    return this.getWalletForLedger("lightning") as LightningWallet;
```

Comment:
I think you should be able to remove the casts here because `AllWallets[K]` is actually the type of the wallet and I think the optionality shouldn't leak out but maybe I am wrong.

Would be nice to get rid of this cast though :)

Reply:
Somehow TS was not happy so I had to add the cast. However, incomit-rs e2e there is no cast! ☹️